graph/memstore/quadstore.go

```diff
@@ -117,17 +116,17 @@ func (qs *QuadStore) ApplyDeltas(deltas []graph.Delta, ignoreOpts graph.IgnoreOp
 		case graph.Add:
 			if !ignoreOpts.IgnoreDup {
 				if _, exists := qs.indexOf(d.Quad); exists {
-					return graph.ErrQuadExists
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadExists}
```

Comment:
Generic quad error type is ok, but we definitely have 3 concrete errors which deserves separate types.